### ISSUE TITLE
Rule Request: let_shorthand

### ISSUE THREAD

**Issue by sadefigu** [realm/SwiftLint#4822]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Optional binding using `if let foo = foo { ... }`, to create an unwrapped variable that shadows an existing optional variable, is an extremely common pattern. This pattern requires the author to repeat the referenced identifier twice, which can cause these optional binding conditions to be verbose, especially when using lengthy variable names.  (copied from [swift evolution](https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md)).

Swift 5.7 introduced let shorthand to help with the above issue:

```swift
let foo: Foo? = ...

if let foo {
    // `foo` is of type `Foo`
}
```

> 1. Why should this rule be added? Share links to existing discussion about what
>    the community thinks about this.

The new rule would make sure that developers are using the let shorthand when appropriate.

It would help to keep the codebase style more consistent. Please see the below references:

-  [swift evolution](https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md)
-  [swift forums](https://forums.swift.org/t/if-let-shorthand/54230)

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

**Would trigger:**

```
if let foo = foo {
```
```
guard let foo = foo {
```

**Would not trigger:**

```
if let foo {
```
```
if let foo = foo() {
```
```
guard let foo = foo() {
```
```
if let foo = foo as? Foo {
```
```
guard let foo = foo as? Foo {
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

This rule does not need to be configurable. 
One could say that we could offer an option to let users choose if they want to enforce the rule for both `if` and `guard` statements, but that may cause the codebase to become less consistent. 

> 4. Should the rule be opt-in or enabled by default? Why?

The rule should be opt-in as it is not a consensus. Teams should be able to decide when and if they want to adopt it.

**Comment by sadefigu** [comment]
This should be closed as a dup of #4202
